router/batchrouter/handle_async.go

```diff
@@ -123,37 +127,25 @@ func (brt *Handle) pollAsyncStatus(ctx context.Context) {
 									}
 									brt.asyncSuccessfulJobCount.Count(len(statusList))
 								} else {
-									failedJobUrl := asyncResponse.FailedJobsURL
-									payload = asyncdestinationmanager.GenerateFailedPayload(destinationsMap[key].Destination.Config, importingList, importId, brt.destType, csvHeaders)
 									startFailedJobsPollTime := time.Now()
 									brt.logger.Debugf("[Batch Router] Fetching Failed Jobs Started for Dest Type %v", brt.destType)
-									failedBodyBytes, statusCode := misc.HTTPCallWithRetryWithTimeout(brt.transformerURL+failedJobUrl, payload, asyncdestinationmanager.HTTPTimeout)
-									brt.logger.Debugf("[Batch Router] Fetching Failed Jobs for Dest Type %v", brt.destType)
+									uploadStatsResp := brt.asyncdestinationmanager.GetUploadStats(GetUploadStatsInput)
 									brt.asyncFailedJobsTimeStat.Since(startFailedJobsPollTime)
 
-									if statusCode != 200 {
-										continue
-									}
-									var failedJobsResponse map[string]interface{}
-									err = json.Unmarshal(failedBodyBytes, &failedJobsResponse)
-									if err != nil {
-										panic("JSON Unmarshal Failed" + err.Error())
-									}
-									internalStatusCode, ok := failedJobsResponse["status"].(string)
-									if internalStatusCode != "200" || !ok {
-										brt.logger.Errorf("[Batch Router] Failed to fetch failed jobs for Dest Type %v with statusCode %v and body %v", brt.destType, internalStatusCode, string(failedBodyBytes))
+									if uploadStatsResp.Status != "200" {
 										continue
 									}
-									metadata, ok := failedJobsResponse["metadata"].(map[string]interface{})
-									if !ok {
-										brt.logger.Errorf("[Batch Router] Failed to typecast failed jobs response for Dest Type %v with statusCode %v and body %v with error %v", brt.destType, internalStatusCode, string(failedBodyBytes), err)
+
+									uploadStatsRespInBytes, _ := stdjson.Marshal(uploadStatsResp)
```

Comment:
Handle the error.

Reply:
added panic, as this is how it was done before. LMK if we need to change anything